ste/sender-blockBlob.go

```diff
@@ -212,6 +220,26 @@ func (s *blockBlobSenderBase) Epilogue() {
 			}
 		}
 	}
+
+	// Upload ADLS Gen 2 ACLs
+	if jptm.FromTo() == common.EFromTo.BlobBlob() && jptm.Info().PreserveSMBPermissions.IsTruthy() {
+		bURLParts := azblob.NewBlobURLParts(s.destBlockBlobURL.URL())
+		bURLParts.BlobName = strings.TrimSuffix(bURLParts.BlobName, "/") // BlobFS does not like when we target a folder with the /
+		bURLParts.Host = strings.ReplaceAll(bURLParts.Host, ".blob", ".dfs")
+		// todo: jank, and violates the principle of interfaces
+		fileURL := azbfs.NewFileURL(bURLParts.URL(), s.jptm.(*jobPartTransferMgr).jobPartMgr.(*jobPartMgr).secondaryPipeline)
+
+		// We know for a fact our source is a "blob".
+		acl, err := s.sip.(*blobSourceInfoProvider).AccessControl()
+		if err != nil {
+			jptm.FailActiveSend("Grabbing source ACLs", err)
+		}
+		acl.Permissions = "" // Since we're sending the full ACL, Permissions is irrelevant.
```

Comment:
Just to double check, the full ACL is always a superset of the permissions?

Reply:
Correct.